chain/src/main/scala-2.11/org/bitcoins/chain/blockchain/Blockchain.scala

```diff
@@ -2,11 +2,10 @@ package org.bitcoins.chain.blockchain
 
 import org.bitcoins.chain.models.BlockHeaderDb
 
-import scala.collection.{mutable, IndexedSeqLike}
+import scala.collection.mutable
 
 /** @inheritdoc */
 case class Blockchain(headers: Vector[BlockHeaderDb])
-    extends IndexedSeqLike[BlockHeaderDb, Vector[BlockHeaderDb]]
```

Comment:
your wrapper classes just `extend Seq` which doesn't make any guarantees about Indexing, so this can devolve to worst case `O(n)` when attempting to do a indexed lookup. As far as I know, Scala doesn't make any guarantees about the implementation type behind this, which is why people never used `Seq` in older versions of Scala. I think things changed in `2.13` with the collections rework, but we don't just support 2.13

Reply:
For instance, this is how `Vector` is defined, you probably want to default to `IndexedSeq` IMO. 

```scala
final class Vector[+A] private[immutable] (private[collection] val startIndex: Int, private[collection] val endIndex: Int, focus: Int)
extends AbstractSeq[A]
   with IndexedSeq[A]
   with GenericTraversableTemplate[A, Vector]
   with IndexedSeqLike[A, Vector[A]]
   with VectorPointer[A @uncheckedVariance]
   with Serializable
   with CustomParallelizable[A, ParVector[A]
```